Modules/Core/FiniteDifference/include/itkDenseFiniteDifferenceImageFilter.hxx

```diff
@@ -41,7 +41,7 @@ DenseFiniteDifferenceImageFilter<TInputImage, TOutputImage>::CopyInputToOutput()
   // Check if we are doing in-place filtering
   if (this->GetInPlace() && this->CanRunInPlace())
   {
-    typename TInputImage::Pointer tempPtr = output.GetPointer();
+    typename TInputImage::Pointer tempPtr = dynamic_cast<TInputImage *>(output.GetPointer());
```

Comment:
Please consider `auto tempPtr = output.GetPointer()` instead.

Reply:
Something like this:

    const void * const inputPixelContainer = input->GetPixelContainer();
    const auto * const tempPtr = output.GetPointer();

    if ((tempPtr != nullptr) && (tempPtr->GetPixelContainer() == inputPixelContainer))
    {
      // the input and output container are the same - no need to copy
      return;
    }

Reply:
Thanks for the suggestion @N-Dekker : I went on to minimally modify the code in a3cca73. Let me know if you think that  `tempPtr` should absolutely be compared to `nullptr` (if not doing it could derive in unexpected situations in some edge cases?) and the convenience of adding the `inputPixelContainer` local variable.

Reply:
@jhlegarreta Does your minimal modification (just changing the declaration of `tempPtr`) compile well?  If so, it's fine by me.

I added the `inputPixelContainer` as a `void` pointer, because I was thinking that `tempPtr->GetPixelContainer() == input->GetPixelContainer()` would not compile, when `tempPtr` and `input` have different pixel types. Just like the comparison `intPtr == floatPtr` fails to compile, for an `int*` and a `float*`: https://godbolt.org/z/sj8snx

The `nullptr` is not _absolutely_ necessary. So again, if your minimal modification solves the problem, it's fine by me.

Reply:
Thanks for the explanations @N-Dekker  !

It compiled for the CI builds (had also tested locally on Windows/MSVC), but just as the existing build error is only reported when ITK is built from the `ITKExamples` `Superbuild`, I prefer to protect myself and avoid further back and forth on this. So done in 57e5f81.